source/gui/settingsDialogs.py

```diff
@@ -62,6 +62,7 @@
 	Set,
 	cast,
 )
+from url_normalize import url_normalize
```

Comment:
we should make this an explicit pip dependency in requirements if we are using it. is there anything inbuilt we can use instead?

Reply:
The closest thing that I am aware of is `urllib.parse`. However, its parsing is not as robust, and does no sanity checking whatsoever. I wrote some custom code based on this, but since `url_normalize` is already 
an implicit dependency, I thought it better to use its more robust checking. Happy to make the dependency explicit - it's MIT licensed so shouldn't represent a problem from a licensing POV.